### PR TITLE
Added documentation on meshlets and how to use them

### PR DESCRIPTION
**Pull request by JakobPer**

### CODE REVIEW

File: docs/meshlets.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    // Perform meshlet division here, and store resulting meshlet in generatedMeshlets
+    return generatedMeshlets;
+}
+```
+
+#### Example That Uses a 3rd Party Library:
+
+An example about how [meshoptimizer](https://github.com/zeux/meshoptimizer) **can** be used via the custom division function that can be passed to `gvk::divide_into_meshlets()`:
+
+Please note: meshoptimizer expects `aMaxIndices` to be divisible by 4. The value 124 has been recommended for Nvidia!
```
Comment (johannesugb) [inline docs/meshlets.md:71]: Please provide references for these two statements (1. divisible by four, 2. 124 recommended for Nvidia)!

Are you sure it is 124 and not 126 (see [Christoph Kubisch - Introduction to Turing Mesh Shaders](https://developer.nvidia.com/blog/introduction-turing-mesh-shaders/)).

If 124 is mentioned somewhere in meshoptimizer, please cite that one!

**Review by johannesugb — CHANGES_REQUESTED** [review]
I have reworked some parts of the text. 
Only issue left: The missing references as commented above.

File: docs/meshlets.md (hunk 1 above)
Comment (JakobPer) [reply docs/meshlets.md:71]: Thanks for the edits.
Sorry bad wording on my part, 124 is the recommendation by meshoptimizer. I'll add that.

Comment (JakobPer) [reply docs/meshlets.md:71]: The divisable by four is not documented in meshoptimizer, I just ran into it while testing.
`assert(max_triangles % 4 == 0); // ensures the caller will compute output space properly as index data is 4b aligned`
This is in the `meshopt_buildMeshlets` function of meshoptimizer.

Comment (johannesugb) [reply docs/meshlets.md]: No need to be sorry, this is just polishing. Thanks for adding the citation.

**Review by johannesugb — APPROVED** [review]

**Pull request merged**